src/cluster_manager.sh

```diff
@@ -0,0 +1,405 @@
+        if [[ "${node_num}" =~ ^[0-9]+$ ]] && [ "${node_num}" -gt "${last_id}" ]; then
+            last_id="${node_num}"
+        fi
+    done
+
+    local -r subnet=$(_get_subnet "${USHIFT_MULTINODE_CLUSTER}")
+    local -r node_id=$((last_id + 1))
+    local -r node_name="${NODE_BASE_NAME}${node_id}"
+    local -r ip_address=$(_get_ip_address "$subnet" "$node_id")
+
+    cluster_healthy
+
+    echo "Creating node: ${node_name}"
+    if ! _add_node "${node_name}" "${USHIFT_MULTINODE_CLUSTER}" "${ip_address}"; then
+        echo "ERROR: failed to create node: ${node_name}" >&2
+        exit 1
+    fi
+    echo "Joining node to the cluster: ${node_name}"
+    if ! _join_node "${node_name}"; then
+        echo "ERROR: failed to join node to the cluster: ${node_name}" >&2
+        echo "=== Add-node log content ===" >&2
+        if sudo podman exec -i "${node_name}" test -f add-node.log; then
+            sudo podman exec -i "${node_name}" cat add-node.log >&2
+        else
+            echo "WARNING: add-node.log not found in ${node_name}" >&2
+        fi
+        exit 1
+    fi
+
+    echo "Node added successfully. To access the new node container, run:"
+    echo "  sudo podman exec -it ${node_name} /bin/bash"
+    return 0
+}
+
+
+cluster_start() {
+    local -r containers=$(_get_cluster_containers)
+
+    if [ -z "${containers}" ]; then
+        echo "ERROR: No cluster containers found" >&2
+        exit 1
+    fi
+
+    echo "Starting cluster"
+    for container in ${containers}; do
+        echo "Starting container: ${container}"
+        sudo podman start "${container}" || true
+    done
+}
+
+
+cluster_stop() {
+    local -r containers=$(_get_running_containers)
+
+    if [ -z "${containers}" ]; then
+        echo "No running cluster containers"
+        return 0
+    fi
+
+    echo "Stopping cluster"
+    for container in ${containers}; do
+        echo "Stopping container: ${container}"
+        sudo podman stop --time 0 "${container}" || true
+    done
+}
+
+
+cluster_destroy() {
+    local containers
+    containers=$(_get_cluster_containers)
+    for container in ${containers}; do
+        echo "Stopping container: ${container}"
+        sudo podman stop --time 0 "${container}" || true
+        echo "Removing container: ${container}"
+        sudo podman rm -f "${container}" || true
+    done
+
+    if sudo podman network exists "${USHIFT_MULTINODE_CLUSTER}"; then
+        echo "Removing podman network: ${USHIFT_MULTINODE_CLUSTER}"
+        sudo podman network rm "${USHIFT_MULTINODE_CLUSTER}" || true
+    fi
+
+    sudo rmmod openvswitch || true
+    delete_topolvm_backend
+
+    echo "Cluster destroyed successfully"
+}
+
+
+cluster_ready() {
+    local -r containers=$(_get_running_containers)
+    if [ -z "${containers}" ]; then
+        echo "No running nodes found"
+        exit 1
+    fi
+    for container in ${containers}; do
+        echo "Checking readiness of container: ${container}"
```

Comment:
```suggestion
        echo "Checking readiness of node: ${container}"
```
Strictly speaking, we're waiting for MicroShift to init, not the container.

Reply:
Done.